tooling/operation/get-parameters-as-json-schema.js

```diff
@@ -15,173 +15,347 @@ const types = {
   header: 'Headers',
 };
 
-function getBodyParam(operation, oas) {
-  const schema = getSchema(operation, oas);
-  if (!schema) return null;
+// This list has been pulled from `openapi-schema-to-json-schema` but been slightly modified to fit within the
+// constraints in which ReadMe uses the output from this library in `@readme/oas-form` as while properties like
+// `readOnly` aren't represented within JSON Schema, we support it within that library's handling of OpenAPI-friendly
+// JSON Schema.
+//
+// https://github.com/openapi-contrib/openapi-schema-to-json-schema/blob/master/index.js#L23-L27
+const unsupportedSchemaProps = [
+  'nullable',
+  // 'discriminator',
+  // 'readOnly',
+  // 'writeOnly',
+  'xml',
+  'externalDocs',
+  'example', // OpenAPI supports `example`, but we're mapping it to `examples` below.
+  'deprecated',
+];
+
+/**
+ * Take a string and encode it to be used as a JSON pointer.
+ *
+ * @link https://tools.ietf.org/html/rfc6901
+ * @param {String} str
+ * @returns {String}
+ */
+function encodePointer(str) {
+  return str.replace('~', '~0').replace('/', '~1');
+}
+
+function isPrimitive(val) {
+  return typeof val === 'string' || typeof val === 'number' || typeof val === 'boolean';
+}
+
+function isPolymorphicSchema(schema) {
+  return 'allOf' in schema || 'anyOf' in schema || 'oneOf' in schema;
+}
+
+/**
+ * Determine if a given schema looks like a `requestBody` schema and contains the `content` object.
+ *
+ * @param {Object} schema
+ * @returns {Boolean}
+ */
+function isRequestBodySchema(schema) {
+  return 'content' in schema;
+}
+
+/**
+ * Given a JSON pointer and an array of examples do a reverse search through them until we find the JSON pointer, or
+ * part of it, within the array.
+ *
+ * This function will allow you to take a pointer like `/tags/name` and return back `buster` from the following array:
+ *
+ * ```
+ *  [
+ *    {id: 20},
+ *    {tags: {name: 'doggie'}}
+ *  ]
```

Comment:
🎆 🎆 🎆 examples! 🎆 🎆 🎆

Reply:
Where does the key `buster` come from in the above example?

Reply:
Just updated this example because it was totally wrong.

Reply:
The pointer should return `doggie` right, not buster?

Reply:
Yeah my bad, will update it again.